src/main.rs

```diff
@@ -165,6 +258,144 @@ fn create_pipeline(app: &App) -> Result<(gst::Pipeline, gtk::Widget), Box<dyn er
     Ok((pipeline, widget))
 }
 
+fn save_settings(
+    folder_button: &gtk::FileChooserButton,
+    options: &gtk::ComboBoxText,
+    timer_entry: &gtk::SpinButton,
+) {
+    let settings = SnapshotSettings {
+        folder: PathBuf::from(folder_button.get_filename()
+                                           .unwrap_or_else(|| {
+                                               glib::get_home_dir().unwrap_or_else(|| {
+                                                   PathBuf::from(".")
+                                               })
+                                           })),
+        format: OutputFormat::from(options.get_active_text()),
+        timer_length: timer_entry.get_value_as_int() as _,
+    };
+    let s = get_settings_file_path();
+    if let Err(e) = serde_any::to_file(&s, &settings) {
+        eprintln!("Error when trying to save file: {:?}", e);
+    } else {
+        println!("Saved settings {:?} in '{}'", settings, s.display());
+    }
+}
+
+fn build_snapshot_settings_window(parent: &gtk::Window) {
+    let s = get_settings_file_path();
+
+    if !s.exists() {
+        if let Some(parent) = s.parent() {
+            if !parent.exists() {
+                if let Err(e) = create_dir_all(parent) {
+                    eprintln!("Error when trying to build settings folder '{}': {:?}",
+                              parent.display(),
+                              e);
+                }
+            }
+        }
+    }
+
+    let settings = if s.exists() && s.is_file() {
+        match serde_any::from_file::<SnapshotSettings, _>(&s) {
+            Ok(s) => s,
+            Err(e) => {
+                eprintln!("Error when opening '{}': {:?}", s.display(), e);
+                SnapshotSettings::default()
+            }
+        }
+    } else {
+        SnapshotSettings::default()
+    };
+
+    //
+    // BUILDING UI
+    //
+    let dialog = gtk::Dialog::new_with_buttons(Some("Snapshot settings"),
+                                               Some(parent),
+                                               gtk::DialogFlags::MODAL,
+                                               &[("Close", gtk::ResponseType::Close.into())]);
```

Comment:
The dialog could use some padding at the window borders :)

Reply:
Yes sir!